content/blog/setting-up-github-actions-for-v.md

```diff
@@ -0,0 +1,355 @@
+list: _making sure everyone is on the same page_. Individuality is important
+for problem solving: it offers new angles, allows people to shine where they
+are good at. But it can be a problem in case of things like formatting
+preferences and build environments. Homogeneity is preferred in these
+instances. Therefore, the decision makers band together to lay down some
+ground rules for everyone to follow. These include things such as:
+
+1. Expected behaviour that is standardized using _tests_.
+2. _Source Code formatting_ guidelines that are strictly enforced.
+3. Managing build and release _environments_ with relevant dependencies managed.
+
+Once all protocols are in place, it is beneficial to automate these steps.
+It saves on time and effort while allowing the team to experiment without
+the fear of breaking stuff. Another benefit is that it allows for faster
+iterations. If something fails, the team will be notified and the regression
+can be taken care of quickly. This is **Continuous Integration** in a
+nutshell.
+
+**Continuous Delivery** and **Continuous Deployment** are concerned with
+automatically building the product after every change and deploying it to
+the staging or production environment, respectively. This allows the
+developers to just write code and push fixes whenever they want. The
+"pipeline" takes care of the rest.
+
+You can read
+[Red Hat's page on CI/CD](https://www.redhat.com/en/topics/devops/what-is-ci-cd)
+for more information. Also, here's their page on the umbrella topic of
+[DevOps](https://www.redhat.com/en/topics/devops).
+
+## When should you consider _not_ setting up a CI/CD environment?
+
+A few years ago, people would argue that setting up a CI environment was
+resource intensive. You would have to setup a CI server, install the
+relevant software on it, setup the notification systems, and put a lot
+of time into general infrastructure management.
+
+Now it's no longer the case. Several providers like
+[GitLab](https://docs.gitlab.com/ee/ci/index.html) and
+[GitHub](https://docs.github.com/en/actions)
+provide first-party CI/CD services. They offer generous free-tiers which
+exceed the requirements of regular developers and small businesses.
+
+If you put in a bit of elbow grease, you can set up your own CI/CD service.
+For example, a popular FOSS pairing is
+[Woodpecker CI](https://woodpecker-ci.org/) with
+[Gitea](https://gitea.io/en-us/).
+
+You still have to write the pipelines for all your repositories, but
+getting started with CI for a project is much easier than it used to be.
+
+## Introduction to GitHub Actions
+
+We will stick to [GitHub Actions](https://docs.github.com/en/actions) for
+this tutorial because it is the most widely used CI/CD service in the
+community at present. We can follow up with instructions for other CI services
+if there's demand for it.
+
+Let us start by discussing some terminology.
+
+A _workflow_ is a set of instructions described in [YAML](https://yaml.org/).
```

Comment:
```suggestion
A _workflow_ is a set of jobs, plus a specification on when to run them, described in [YAML](https://yaml.org/).
```

Reply:
I'm sorry I read the suggestion the other way around.